test/e2e/specs/editor/blocks/list.spec.js

```diff
@@ -0,0 +1,872 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'List', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'can be created by using an asterisk at the start of a paragraph block', async ( {
+		editor,
+		page,
+	} ) => {
+		// Create a block with some text that will trigger a list creation.
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+		await page.keyboard.type( '* A list item' );
+
+		// Create a second list item.
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Another list item' );
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:list -->
+<ul><li>A list item</li><li>Another list item</li></ul>
+<!-- /wp:list -->`
+		);
+	} );
```

Comment:
Now that we have `expect.poll` available, I think we can write this instead to make it auto-wait for the result to reduce flakiness.

```suggestion
		await expect.poll( editor.getEditedPostContent ).toBe(
			`<!-- wp:list -->
<ul><li>A list item</li><li>Another list item</li></ul>
<!-- /wp:list -->`
		);
	} );
```

Same for the other assertions below.